full-service/src/service/t3_sync.rs

```diff
@@ -0,0 +1,152 @@
+                        .get_pooled_conn()
+                        .expect("failed getting wallet db connection");
+
+                    loop {
+                        if thread_stop_requested.load(Ordering::SeqCst) {
+                            log::debug!(logger, "T3SyncThread stop requested.");
+                            break;
+                        }
+
+                        match sync_txos(conn, &t3_connection, &logger) {
+                            Ok(()) => (),
+                            Err(e) => log::error!(&logger, "Error during t3 sync:\n{:?}", e),
+                        }
+
+                        // This sleep is to allow other API calls that need access to the database a
+                        // chance to execute, because the t3 sync process requires a write lock on
+                        // the database.
+                        thread::sleep(std::time::Duration::from_millis(T3_SYNC_INTERVAL));
+                    }
+                    log::debug!(logger, "T3SyncThread stopped.");
+                })
+                .expect("failed starting t3 sync thread"),
+        );
+
+        Self {
+            join_handle,
+            stop_requested,
+        }
+    }
+
+    pub fn stop(&mut self) {
+        self.stop_requested.store(true, Ordering::SeqCst);
+        if let Some(join_handle) = self.join_handle.take() {
+            join_handle.join().expect("T3SyncThread join failed");
+        }
+    }
+}
+
+impl Drop for T3SyncThread {
+    fn drop(&mut self) {
+        self.stop();
+    }
+}
+
+pub fn sync_txos(
+    conn: Conn,
+    t3_connection: &T3Connection,
+    logger: &Logger,
+) -> Result<(), T3SyncError> {
+    // get all txos from the database that haven't synced yet to t3 and that have an
+    // authenticated sender memo
+    log::debug!(logger, "Syncing txos to t3");
+
+    let txos = Txo::get_txos_that_need_to_be_synced_to_t3(Some(TXO_CHUNK_SIZE), conn)?;
+
+    for txo in txos {
+        let txo_memo = txo.memo(conn)?;
+        let memo = match txo_memo {
+            TxoMemo::AuthenticatedSender(memo) => memo,
+            _ => continue,
+        };
+
+        sync_txo(&txo, &memo, t3_connection)?;
+        txo.update_is_synced_to_t3(true, conn)?;
+    }
+
+    Ok(())
+}
+
+fn sync_txo(
+    txo: &Txo,
+    memo: &AuthenticatedSenderMemo,
+    t3_connection: &T3Connection,
+) -> Result<(), T3SyncError> {
+    let mut transparent_transaction = TransparentTransaction::new();
+
+    let sender_address_hash_bytes = hex::decode(&memo.sender_address_hash)?;
+    transparent_transaction.set_sender_address_hash(sender_address_hash_bytes);
+
+    transparent_transaction.set_token_id(txo.token_id as u64);
+    transparent_transaction.set_amount(txo.value as u64);
+
+    let public_key_bytes = txo.public_key()?.as_bytes().to_vec();
+    transparent_transaction.set_public_key_hex(hex::encode(public_key_bytes));
+
+    let _ = t3_connection.create_transaction(transparent_transaction)?;
```

Comment:
In order to have parity with Moby we should also be filling in either `sender_institution` or `recipient_institution`. The institution enum value should be provided as a config arg (like the t3 api key) and then filled in either sender or recipient depending on if t3 is sending or receiving the transaction. If no enum is provided then `NONE` institution should be selected.

Reply:
Could we just reference that based on the api key and information stored in the t3 database? It seems that for most use cases it wouldn't be applicable, only to the institutions that are established as part of the enum

Reply:
T3 server won't know if it was the sender or recipient address which belongs to the full service node. We could store the enum in the t3 server and return it to full service with some RPC but it would still be up to full service to assign the enum value to the correct field.